protocol/messenger_handler.go

```diff
@@ -1575,16 +1591,33 @@ func (m *Messenger) HandleChatMessage(state *ReceivedMessageState) error {
 			m.logger.Info("not handling contact message since clock lower")
 			return nil
 		}
-		receivedMessage.ContactRequestState = common.ContactRequestStatePending
-		contact.ContactRequestClock = receivedMessage.Clock
-		contact.ContactRequestReceived()
-		state.ModifiedContacts.Store(contact.ID, true)
-		state.AllContacts.Store(contact.ID, contact)
-		err = m.createContactRequestNotification(state.CurrentMessageState.Contact, state, receivedMessage)
-		if err != nil {
-			return err
-		}
+		currentContact, ok := state.AllContacts.Load(contact.ID)
+		if ok && currentContact.Added {
+			// We've already added this contact, so let's send an ACK
+			ensName, err := m.settings.ENSName()
+			if err != nil {
+				return err
+			}
 
+			displayName, err := m.settings.DisplayName()
+			if err != nil {
+				return err
+			}
+			_, err = m.sendContactUpdate(context.Background(), currentContact.ID, displayName, ensName, "")
```

Comment:
@cammellos in https://github.com/status-im/status-desktop/issues/8702#issuecomment-1369793368 you've mentioned that contacts should send an ACK response which would add one byte.

I"m not sure where those byte would be added, as we would just send a contact update (which is interpreted as the request being accepted).

If this is not at all what you were looking for, please let me know. I just took a first stab on this now.

Reply:
@0x-r4bbit I meant something a bit different, the most economical way to do it, would be to piggy back a field on whether you have added the other to contact to the datasync ACK ( https://github.com/vacp2p/mvds/blob/d6b27d3843ffae66806cc969596af342c6e7a839/protobuf/sync.proto#L7 ) for example. (That could literally be a single bit 0/1 whether we have the pk of the user in contacts)

Modifying datasync it's a bit of a pain but I guess worth considering, the solution here is probably on the heavier side, since we are increasing considerably the number of messages, while modifying datasync would not change the number of messages in the network, just make them slightly large, probably we should have a discussion about potential approaches, as there could be better solutions